core/trino-main/src/test/java/io/trino/sql/query/TestRowFilter.java

```diff
@@ -486,4 +495,64 @@ public void testInsert()
+                new QualifiedObjectName(MOCK_CATALOG, "tiny", "nation_with_hidden_column"),
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "nationkey < 1"));
+
+        assertions.query("SELECT * FROM mock.tiny.nation_with_hidden_column")
+                .assertThat()
+                .skippingTypesCheck()
+                .matches("VALUES (BIGINT '0', 'ALGERIA', BIGINT '0', ' haggle. carefully final deposits detect slyly agai')");
+        // TODO https://github.com/trinodb/trino/issues/10005 - support insert into a table with hidden columns
+        assertThatThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation_with_hidden_column VALUES (101, 'POLAND', 0, 'No comment')"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("scope: 5, fields mappings: 4");
+        // TODO https://github.com/trinodb/trino/issues/10005 - support insert into a table with hidden columns
+        assertThatThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation_with_hidden_column VALUES (0, 'POLAND', 0, 'No comment')"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("scope: 5, fields mappings: 4");
+        assertThatThrownBy(() -> assertions.query("UPDATE mock.tiny.nation_with_hidden_column SET name = 'POLAND'"))
+                .isInstanceOf(TrinoException.class)
+                .hasMessageContaining("Updating a table with a row filter is not supported");
+        assertions.query("DELETE FROM mock.tiny.nation_with_hidden_column WHERE regionkey < 5")
+                .assertThat()
+                .skippingTypesCheck()
+                .matches("SELECT BIGINT '1'");
+    }
+
+    @Test
+    public void testRowFilterOnHiddenColumn()
+    {
+        accessControl.reset();
+        accessControl.rowFilter(
+                new QualifiedObjectName(MOCK_CATALOG, "tiny", "nation_with_hidden_column"),
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "\"$hidden\" < 1"));
+
+        assertions.query("SELECT count(*) FROM mock.tiny.nation_with_hidden_column")
+                .assertThat()
+                .skippingTypesCheck()
+                .matches("VALUES BIGINT '25'");
+        // TODO https://github.com/trinodb/trino/issues/10006 - support insert into a table with row filter that is using hidden columns
+        assertThatThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation_with_hidden_column VALUES (101, 'POLAND', 0, 'No comment')"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("scope: 5, fields mappings: 4");
+        // TODO https://github.com/trinodb/trino/issues/10006 - support insert into a table with row filter that is using hidden columns
+        assertThatThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation_with_hidden_column VALUES (0, 'POLAND', 0, 'No comment')"))
```

Comment:
Repeated case

Reply:
`nationkey` is irrelevant. Row filter here is not using `nationkey` column at all.

Reply:
Ah, indeed. But the one in `testRowFilterWithHiddenColumns` was also removed and it wasn't repeated?

Reply:
Or am I imagining things, hm

Reply:
> Or am I imagining things, hm

Looks like I am, sorry about that :)